module/Company/src/Company/Mapper/Company.php

```diff
@@ -31,8 +31,61 @@ public function __construct(EntityManager $em)
     {
         $this->em = $em;
     }
+    /**
+     * Saves all unsaved entities, that are marked persistent
+     *
+     */
+    public function save()
+    {
+        $this->em->flush();
+    }
 
     /**
+     * Delete the company identified with $slug
+     *
+     * @param mixed $slug
+     */
+    public function deleteWithSlug($slug)
```

Comment:
If I "throw out a box with its contents", then I'm throwing out the box including its contents. I'm not throwing out the box based on its contents.